### PR TITLE
Optimized the snap

### PR DESCRIPTION
**Pull request by soumyaDghosh**
1. Using gnome-sdk for every gnome related libraries
2. Reduces the file size
3. Using lzo for fast startup

Additionally, you can simply connect this repo to the snapcraft builder in the [website](https://snapcraft.io/celeste/builds), you can build all the possible architectures[arm64, amd64, armhf]. Also, you don't need to depend on github workflows to push the snap. Just keep the versioning in mind. You just need to click the `Trigger New Build` button properly.

### CODE REVIEW

**Comment by hwittenborn** [comment]
Thank you for the PR @soumyaDghosh! I'm not sure how using the `gnome-sdk` will work (last time I checked, the version of GNOME's libraries in Ubuntu 22.04 weren't new enough - though I'm not sure how the `gnome-sdk` works in relation to that), but I saw in the Discord channel that you tested it so I'm assuming it's fine. I'll do some testing tomorrow (it's quite late at my place right now) just to make sure everything works and then we can go from there.

> Additionally, you can simply connect this repo to the snapcraft builder in the [website](https://snapcraft.io/celeste/builds), you can build all the possible architectures[arm64, amd64, armhf].

The Prebuilt-MPR only supports `amd64` and (experimentally at the moment) `arm64`, which would cause an issue with `armhf` being used. As far as I know `armhf` is being phased out in favor of `arm64` though, so I'll probably end up using that instead.

I'd also like to note that I plan on making the Snap and Flatpak both pull from the AppImage once that gets implemented (see https://github.com/hwittenborn/celeste/issues/16). It'll probably create a bit bigger of a binary size, but it's more maintainable by me in the long run, and helps me to not duplicate things across different workflow files. The Flatpak currently just pulls from the Snap, so it'll just make it so that both pull from a singular source that I manage.

Do you have any opinions on that matter? I've been pretty set on making this all pull from the Snap, but I'm open to feedback in that area.

**Comment by soumyaDghosh** [comment]
For snap, it's not the best option. If this was an electron, I wouldn't have said that, but as it's a native app, currently written in rust. This is the best method. 

For the prebuild mpr, I can make it to build from source, so, no problem in that. 

So, IMO, keep the snap as it is. I'll help you maintain it, if you need. It's in a more controlled state now in terms of debugging and dependency.

**Comment by soumyaDghosh** [comment]
Kindly merge this PR.

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -24,34 +16,24 @@ parts:
...
+      DESTDIR=$CRAFT_PART_INSTALL just install
+
+  deps:
+    after:
+      - celeste
+    plugin: nil
+    stage-packages:
+      - rclone
+    prime:
+      - usr/bin/rclone
```
Comment (hwittenborn) [inline snap/snapcraft.yaml:36]: Is there a reason we're doing this all as a separate step instead of just doing it in the previous one?

**Review by hwittenborn — CHANGES_REQUESTED** [review]

File: snap/snapcraft.yaml (hunk 1 above)
Comment (soumyaDghosh) [reply snap/snapcraft.yaml:36]: So, that later we can simply add other packages, without cluttering the main part.

Comment (hwittenborn) [reply snap/snapcraft.yaml:36]: I feel like the `snapcraft.yaml` file is still quite legible even with it all in one section. Does it affect the resulting package in any meaningful way? If it doesn't I'd probably want to stick with it being all in one section.

**Comment by soumyaDghosh** [comment]
Now you can push these changes @hwittenborn

**Review by hwittenborn — APPROVED** [review]

**Review by hwittenborn — APPROVED** [review]
Thanks for the contribution! :)

**Pull request merged**